### PR TITLE
[LHDM-1092] Add fully controlled option for Header

### PR DESCRIPTION
**Pull request by seesleestak**
## Summary

- Added `isMenuOpen` and `onMenuToggle` props to `Header` to allow for fully controlled operation of the `Header` state.
- [LHDM-1092 ticket](https://jira.cms.gov/browse/LHDM-1092)

## How to test

1. `yarn install`
2. `yarn storybook:healthcare`
3. Take a look at "Logged In Controlled Header" story for `Header`

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`

### If this is a change to code:

- [x] Created or updated unit tests to cover the logic added
- [ ] Updated unit test snapshots (`yarn update-snapshots`) and loki reference images **if necessary** (`yarn loki && yarn loki:healthcare && yarn loki:medicare`)

### CODE REVIEW

**Comment by seesleestak** [comment]
Open to feedback on anything here: naming, conventions, testing, etc.

File: packages/ds-healthcare-gov/src/components/Header/Header.test.jsx [hunk 1]
```diff
@@ -132,4 +132,16 @@ describe('Header', function () {
...
+  it('toggles open menu for fully controlled operation', () => {
+    const onMenuToggle = jest.fn();
+    const wrapper = render({
+      isMenuOpen: false,
+      onMenuToggle,
+    });
+    const actionMenuProps = () => wrapper.find('ActionMenu').props();
+    expect(actionMenuProps().open).toBe(false);
+    actionMenuProps().onMenuToggleClick();
+    expect(onMenuToggle).toHaveBeenCalled();
```
Comment (pwolfert) [inline packages/ds-healthcare-gov/src/components/Header/Header.test.jsx:145]: Thanks for adding tests!

**Review by pwolfert — APPROVED** [review]
Thanks for contributing this!

I updated the Storybook story to make it interactive while keeping it controlled.

**Comment by pwolfert** [comment]
Note that our CI tests are currently broken. When https://github.com/CMSgov/design-system/pull/2155 is merged, we can re-run the tests.

**Comment by seesleestak** [comment]
> I updated the Storybook story to make it interactive while keeping it controlled.

Cool, thanks! I'd played around with making it truly interactive, but it's been a minute since I've worked with storybook...

**Pull request merged**